qemu/tests/virtio_serial_throttling.py

```diff
@@ -0,0 +1,64 @@
+from avocado.utils import process
+from virttest import error_context
+from virttest.utils_virtio_port import VirtioPortTest
+from qemu.tests.virtio_serial_empty_line import get_device_name_of_port, get_port_info, check_port_info
+import socket
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Induce throttling by opening chardev but not reading data from it:
+    1) Start guest with virtio-serial with unix option
+    2) On the host, open the chardev but don't read from it
+    3) In the guest, write to the virtio-serial port.
+    4) repeat step 1 to 3 with tcp option.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_chardev_params(vm, serial_id):
+        serial_device = vm.devices.get(serial_id)
+        chardev_id = serial_device.get_param('chardev')
+        chardev_device = vm.devices.get(chardev_id)
+        chardev_params = chardev_device.params
+        return chardev_params
+    
+    def get_serial_params(vm, serial_id):
+        return vm.devices.get(serial_id)
```

Comment:
The return value is a serial device object, not serial params.
`serial_name = vm.devices.get(serial_id).get_param('name')` is ok.
or you can combine this with the function `get_chardev_params`, anyone is ok for me.